tasks/main.yml

```diff
@@ -1,4 +1,35 @@
 ---
+- name: Update apt cache (Debian derivatives)
+  become: true
+  ansible.builtin.apt:
+    cache_valid_time: "{{ gxadmin_apt_cache_valid_time }}"
+  when: >-
+    ansible_os_family == "Debian"
+    and gxadmin_manage_dependencies
+    and gxadmin_manage_apt_cache
+
+- name: Ensure role and gxadmin dependencies are installed
+  become: true
+  ansible.builtin.package:
+    name:
+      - git
+      - make
+    state: present
+  when: gxadmin_manage_dependencies
```

Comment:
1. Why are you doing this with `apt` rather than preferred `package`?
2. I don't love having a flag for manaing the apt cache, that's so weirdly debian specific and not relevant for "download a simple package"

Reply:
I am using [`ansible.builtin.apt`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_module.html) because updating the APT cache is Debian-specific. [`ansible.builtin.package`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html) does not update the repository metadata.

I think we do not need to do the same for EL because [it is smarter](https://discussion.fedoraproject.org/t/dnf-updates-repository-cache-whenever-i-try-to-install-a-package/76627) than APT in this regard.

I agree with (2) and would also prefer not having this thing in the role variables. Let's say whatever I did, there was a 50% chance that you would disagree, and removing lines is easier than adding them haha

I proceed to remove `gxadmin_manage_apt_cache` and `gxadmin_apt_cache_valid_time`.

Reply:
Yes, I'm aware apt cache is debian specific, and the differences between the two.

However, the point was the rationale for forcing a cache update for installing `git`: A cache update should only be required in cases of adding a new repository, where the packages in the newly added repo won't be known to the cache.  (Or in cases of e.g. a docker container where the cache can be ancient, but this isn't relevant for 99% of users, and everyone starts their docker containers with a big 'update the cache' stanza anyway.) Since we're not adding a new apt repository, it thus shouldn't be needed.

Reply:
Well, why should this role not work on the [official Ubuntu Docker image](https://hub.docker.com/_/ubuntu)...? In general it is unknown whether the cache is ancient or not. That's why I said DNF is smarter :sweat_smile:.